src/Menu/CachedBackendMenuBuilder.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bolt\Menu;
+
+use Psr\SimpleCache\CacheInterface;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+class CachedBackendMenuBuilder
+{
+    /** @var CacheInterface */
+    private $cache;
+
+    /** @var BackendMenuBuilder */
+    private $menuBuilder;
+
+    /** @var Stopwatch */
+    private $stopwatch;
+
+    public function __construct(CacheInterface $cache, BackendMenuBuilder $menuBuilder, Stopwatch $stopwatch)
+    {
+        $this->cache = $cache;
+        $this->menuBuilder = $menuBuilder;
+        $this->stopwatch = $stopwatch;
+    }
+
+    public function getMenu(): array
```

Comment:
rename to buildMenu, because it is a builder class (change it in every builder class)

Reply:
Done